lib/battle/DamageCalculator.h

```diff
@@ -25,8 +26,10 @@ class DLL_LINKAGE DamageCalculator
 {
 	const CBattleInfoCallback & callback;
 	const BattleAttackInfo & info;
+    static std::unique_ptr<LuaExpressionEvaluator> attackSkillEvaluator;
+    static std::unique_ptr<LuaExpressionEvaluator> defenseSkillEvaluator;
```

Comment:
Static lua states will not work when accessed from multiple threads. And BattleAI does uses multithreading. And creating lua state every time would be too costly...

Need to investigate what lua offers for concurrency.

Reply:
how many threads will game use for damage calculator？

Reply:
Why you're asking? 
As many as player has cores in CPU, possibly more.

Reply:
Might actually be interesting idea - allocate N states, and then search for a free one. But probably overkill, at least for now.

So let's start with simple and straightforward solution - protect this class with mutex

Reply:
in that solution, should I merge setVariable() and evaluate() into one function?

Reply:
Yeah, makes sense

Reply:
ok